[GCC 11] DataFormats/Candidate/interface/Candidate.h: error: 'this' pointer is null

Log file: https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_amd64_gcc11/CMSSW_12_1_X_2021-07-30-0900/DQMOffline/JetMET
Error message: 
```
In member function 'reco::Candidate::const_iterator reco::Candidate::end() const',
    inlined from 'bool PFJetIDSelectionFunctor::firstDataCuts(const reco::Jet&, pat::strbitset&, PFJetIDSelectionFunctor::Version_t)' at /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h:415:84:
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/DataFormats/Candidate/interface/Candidate.h:145:79: error: 'this' pointer is null [-Werror=nonnull]
   145 |     const_iterator end() const { return const_iterator(this, numberOfDaughters()); }
      |                                                              ~~~~~~~~~~~~~~~~~^~
In file included from /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/DataFormats/Candidate/interface/LeafCandidate.h:11,
                 from /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/DataFormats/Candidate/interface/CompositePtrCandidate.h:3,
                 from /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/DataFormats/JetReco/interface/Jet.h:17,
                 from /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/DQMOffline/JetMET/interface/JetAnalyzer.h:36,
                 from /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/DQMOffline/JetMET/src/JetAnalyzer.cc:15:
In member function 'reco::Candidate::const_iterator reco::Candidate::end() const',
    inlined from 'bool PFJetIDSelectionFunctor::firstDataCuts(const reco::Jet&, pat::strbitset&, PFJetIDSelectionFunctor::Version_t)' at /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h:415:84:
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/3f7c5ca595fd8634b26b919adc27f210/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-07-30-0900/src/DataFormats/Candidate/interface/Candidate.h:145:79: error: 'this' pointer is null [-Werror=nonnull]
   145 |     const_iterator end() const { return const_iterator(this, numberOfDaughters()); }
      |                                                              ~~~~~~~~~~~~~~~~~^~
```

Not sure if the error is real (it doesn't complain about `begin()` or about `iterator end()`)

--- a/PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h
+++ b/PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h
@@ -412,7 +412,7 @@ public:  // interface
       double e_muf = 0.0;
       nch = 0;
       nconstituents = 0;
-      for (reco::Jet::const_iterator ibegin = basicJet->begin(), iend = patJet->end(), isub = ibegin; isub != iend;
+      for (reco::Jet::const_iterator ibegin = basicJet->begin(), iend = basicJet->end(), isub = ibegin; isub != iend;
            ++isub) {
         reco::PFJet const *pfsub = dynamic_cast<reco::PFJet const *>(&*isub);
         e_chf += pfsub->chargedHadronEnergy();

